modules/chanserv/unban_self.c

```diff
@@ -49,6 +49,12 @@ cs_cmd_unban(struct sourceinfo *si, int parc, char *parv[])
 		return;
 	}
 
+	if (metadata_find(mc, "private:close:closer"))
+	{
+		command_fail(si, fault_noprivs, STR_CHANNEL_IS_CLOSED, channel);
+		return;
+	}
+
```

Comment:
This needs to be after the block 3 levels below it (to the old line 70); permissions checks should come before closure checks.